Stripe/STPPaymentContext.m

```diff
@@ -424,9 +424,9 @@ - (void)shippingAddressViewController:(STPShippingAddressViewController *)addres
     [self.delegate paymentContextDidChange:self];
     [self appropriatelyDismissViewController:addressViewController completion:^{
         if (self.state == STPPaymentContextStateRequestingPayment) {
-            self.state = STPPaymentContextStateNone;
             [self requestPayment];
         }
+        self.state = STPPaymentContextStateNone;
```

Comment:
Should be an `else` here, or otherwise you might clobber some state set up in request payment